scripts/release/prepare-release.sh

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2025 The Jaeger Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+# Automated release script for Jaeger
+# This script automates the manual steps in the release process
+
+set -euo pipefail
+
+# Source utility functions
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${SCRIPT_DIR}/utils.sh"
+
+# Configuration
+REPO="jaegertracing/jaeger"
+DRY_RUN=false
+
+usage() {
+    cat << EOF
+Usage: $0 [OPTIONS]
+
+Automated release script for Jaeger
+
+OPTIONS:
+    -d, --dry-run     Run in dry-run mode (no actual changes)
+    -h, --help        Show this help message
+
+EOF
+}
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        -d|--dry-run)
+            DRY_RUN=true
+            shift
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        *)
+            log_error "Unknown option: $1"
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+# Main function
+main() {
+    log_info "Starting automated release process..."
+    
+    # Validate repository setup
+    validate_repository
+    
+    # Check if GitHub CLI is installed and authenticated
+    if ! command -v gh > /dev/null 2>&1; then
+        log_error "GitHub CLI (gh) is not installed. Please install it first."
+        exit 1
+    fi
+    
+    # Check if gh is authenticated
+    if ! gh auth status > /dev/null 2>&1; then
+        log_error "GitHub CLI is not authenticated. Please run 'gh auth login' first."
+        exit 1
+    fi
+    
+    # Get current versions
+    log_info "Getting current versions..."
+    read -r current_version_v1 current_version_v2 <<< "$(get_current_versions)"
```

Comment:
the versions should be given as input and determined by start.sh, no need to re-derive them here. See description in #7500